src/notation/internal/notationparts.cpp

```diff
@@ -1263,26 +1320,33 @@ void NotationParts::updateSoloist(const PartInstrumentList& parts)
     }
 }
 
-void NotationParts::sortParts(const PartInstrumentList& parts)
+void NotationParts::sortParts(const PartInstrumentList& parts, const QList<Ms::Staff*>& originalStaves)
 {
     Q_ASSERT(score()->parts().size() == static_cast<int>(parts.size()));
 
-    for (int i = 0; i < parts.size(); ++i) {
-        const Part* currentPart = score()->parts().at(i);
+    QList<int> staffMapping;
+    QList<int> trackMapping;
+    int idx2 = 0;
+    bool sort = false;
 
-        if (currentPart->id() == parts.at(i).partId) {
-            continue;
+    int partIndex = 0;
+    for (const PartInstrument& pi: parts) {
+        Ms::Part* currentPart = pi.isExistingPart ? part(pi.partId) : masterScore()->parts()[partIndex];
```

Comment:
I think you need to check that masterScore()->parts() contains partIndex, because we can get SIGSEVG here

Like this:

QList<Part*> masterParts = masterScore()->parts()
for (const PartInstrument& pi: parts) {
if (partIndex >= masterParts.size()) {
...
}
...
}

Reply:
Good catch :+1:
This is supposed to be checked by the assert in line 1325 but there `score()` is used instead of `masterScore()`. Until now this is not a problem since `sortParts()` is called for main scores only. But when this method is used for sorting parts in an excerpt, then `score()` and `masterScore()` are no longer the same. Also the score in `sortParts()` should be same as the score of the passed staves.
So, instead of using `score()` or `masterScore()` and testing whether this score is the same as (the first stave of) `originalStaves`, `originalstaves[0]->score()` could be used and check on empty `origialStaves` only.

Reply:
Done.